controllers/maker-controller.js

```diff
@@ -1,38 +1,52 @@
 const express = require('express');
 const _ = require('lodash');
+const jwt = require('express-jwt');
 const { App } = require('../db/models');
 const { makeDocument, getDocument } = require('../common/lib/eversign');
 const { createVerification } = require('../common/lib/jumio');
 
 const Router = express.Router();
 
+Router.use(jwt({ secret: process.env.JWT_SECRET }));
+
 Router.use(async (req, res, next) => {
   try {
-    const { accessToken } = req.query;
-    const app = await App.findOne({
-      where: { accessToken },
+    const { user } = req;
+    if (!user || !user.data.username) {
+      return res.status(400).json({ success: false });
+    }
+    const { username } = user.data;
+    const apps = await App.findAll({
+      where: { adminBlockstackID: username },
       attributes: {
         exclude: ['status', 'notes'],
       },
     });
-    if (app) {
+
+    if (req.query.appId) {
+      const app = apps.find((_app) => String(_app.id) === req.query.appId);
+      if (!app) {
+        return res.status(404).json({ success: false });
+      }
       req.app = app;
+    }
+
+    if (apps.length) {
+      req.apps = apps;
       return next();
     }
```

Comment:
@hstove Only just encountered this, as think I've always had apps. We don't want to return `400` when someone has no apps, rather we should see they have none, right? i.e. return `[]`, right?

Reply:
yeah thats a valid point :+1: